pkg/platform/runtime2/setup/setup.go

```diff
@@ -113,20 +117,57 @@ func NewWithModel(target Targeter, msgHandler MessageHandler, model ModelProvide
 	return &Setup{model, target, msgHandler, nil}
 }
 
-// Update installs the runtime locally (or updates it if it's already partially installed)
+// updateStepError attaches a label to an error returned during the Update() function.
+// The label will be used in the analytics event send during setup failures.
```

Comment:
This is too much complexity for something we care very little about. Let's keep it simple, just make the label more generic.

Reply:
I liked this approach, as it ensures that every error in the `update()` method triggers an analytics event.  But are you suggesting to just send one label for any error that the `update()` method returns?

Reply:
Yes, for now I think that's fine. The reason we put it in the analytics is specifically because it was cheap to do so. It's already a redundancy because the actual error should be reported and analyzed on rollbar, not GA.

So any sort of hoop jumping to get this into the GA label is too much effort.

Reply:
I noticed that we would still do double reporting.  If you look at the deploy runner, you will see, that there are four occurrences of `runtime.New()` each of which will send a `"runtime", "start"` event.  

To do this cleanly (and not with lots of hard-to-grok) state variables and if-clauses inside the runtime package, we probably need a separate struct (maybe `RuntimeAnalytics`) that we instantiate somewhere at the runner level, and then can forward to `runtime.New()` and `runtime.Update()` calls. This struct could ensure that every Event is sent only once during an instances lifetime.

Another option would be to add `analytics.EventOnce()` and `analytics.EventWithLabelOnce` calls that keep track of what events have been reported already (in some struct).  That is more magical though.

I could do either of these things (or something else) in this PR, or create a follow-up story... If it was up to me, I would just go for the first approach in this PR.  But I let you decide, of course.

Reply:
Yeah I think what we want is an `analytics.New()` that returns a struct which can be configured to only accept (certain?) events once. It'd be nice if all of the complexities regarding analytics are contained within the analytics package.

Reply:
That's a separate story altogether though, like we discussed in stand.